common/low_pass_filter.c

```diff
@@ -1,5 +1,29 @@
 #include "low_pass_filter.h"
+#include <stddef.h>
 
-w_status_t low_pass_filter_init(void) {
-    return W_FAILURE;
+// Initialize the low-pass filter
+w_status_t low_pass_filter_init(double *alpha, double response_time) {
+    if (alpha == NULL || response_time <= 0) {
+        return W_INVALID_PARAM; // Return specific error for invalid parameters
+    }
+    // Calculate and store the alpha value using the low_pass_alpha function
+    *alpha = low_pass_alpha(response_time, response_time);
+    return W_SUCCESS; // Return success after initialization
 }
+
+// Update the low-pass filter with a new value
+w_status_t update_low_pass(double *alpha, uint16_t new_value, double *low_pass_value) {
```

Comment:
imo a better name would be like `new_input` or new_input_value, since new_value could be referring to the new calculated value (the docs does make it clear, but clarity shld be built into names too)